[SYCL] Rename project to oneAPI DPC++ Compiler (#1249)
intel/llvm: README.md

Intel staging area for llvm.org contribution.
Home for Intel LLVM-based projects:

>  - oneAPI Data Parallel C++ compiler - see **sycl** branch. More information on oneAPI and DPC++ is available at ([https://www.oneapi.com/](https://www.oneapi.com/))

Reviewer: Could you re-format the text in this document to keep 80-character limit, please?

Author: np

## License
See [LICENSE.txt](sycl/LICENSE.TXT) for details.

## Contributing
See [CONTRIBUTING.md](CONTRIBUTING.md) for details.

## Sub-projects Documentation
 - oneAPI Data Parallel C++ compiler - See [GetStartedGuide.md](sycl/doc/GetStartedGuide.md)

## DPC++ extensions

> DPC++ is an open, cross-architecture language built upon the ISO C++ and Khronos SYCL* standards. DPC++ extends these standards with a few extensions. See [sycl/doc/extensions](sycl/doc/extensions) for more details.

Reviewer: I think we might have more than "a few" extensions, so maybe it's better to rephrase to something like:
```suggestion
DPC++ extends these standards with a number of extensions, which can be found in [sycl/doc/extensions](sycl/doc/extensions) directory.
```

Author: Looks better indeed. Thanks.
